warning: unknown warning group

I use LinuxMint 19, and
```.log
/usr/bin/c++ -v
clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
```
and get:
```.log
make
Scanning dependencies of target uncrustify
[  2%] Building CXX object CMakeFiles/uncrustify.dir/src/align.cpp.o
In file included from /home/guy/Software/uncrustify/src/align.cpp:10:
In file included from /home/guy/Software/uncrustify/src/align.h:11:
In file included from /home/guy/Software/uncrustify/src/uncrustify_types.h:16:
In file included from /home/guy/Software/uncrustify/src/options.h:13:
In file included from /home/guy/Software/uncrustify/src/option.h:14:
/home/guy/Software/uncrustify/src/enum_flags.h:16:32: warning: unknown warning group
      '-Wunused-but-set-parameter', ignored [-Wunknown-warning-option]
#pragma GCC diagnostic ignored "-Wunused-but-set-parameter"
                               ^
1 warning generated.
```

--- a/src/enum_flags.h
+++ b/src/enum_flags.h
@@ -9,9 +9,9 @@
 #ifndef ENUM_FLAGS_H_INCLUDED
 #define ENUM_FLAGS_H_INCLUDED
 
-#if __GNUC__ == 4
+#if __GNUC__ == 4 && !defined (__clang__)
 #pragma GCC diagnostic push
-#if __GNUC_MAJOR__ < 9 || __GNUC_PATCHLEVEL__ < 2
+#if __GNUC_MINOR__ < 9 || __GNUC_PATCHLEVEL__ < 2
 // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=59624
 #pragma GCC diagnostic ignored "-Wunused-but-set-parameter"
 #endif
@@ -89,7 +89,7 @@ protected:
 
 } // namespace uncrustify
 
-#if __GNUC__ == 4
+#if __GNUC__ == 4 && !defined (__clang__)
 #pragma GCC diagnostic pop
 #endif
 

